### DISCUSSION TITLE
To Automate the tool like image editor with cypress

### DISCUSSION THREAD

**Discussion by Javed0005** [cypress-io/cypress#28355] [Questions and Help]
Hi,

I hope you're doing well. I've been working on automating an image editor using tools like AutoIT, SikuliX, and AutoHotkey. I'm attempting to integrate these tools with Cypress using the cy. exec command by providing the path to the script file. However, I'm facing challenges, and I would appreciate any guidance or insights you can offer.

Objective:
Automate interactions with an image editor that does not expose elements in the DOM for traditional web automation. Unfortunately, I cannot use XPath or CSS selectors, as inspecting the elements does not reveal them.

Current Approach:
I'm using cy.exec('path/to/your/script.exe', { log: true }) to execute external scripts. The scripts work independently, but the integration with Cypress is not producing the desired results.

Challenges:

The image editor doesn't expose elements in the DOM, making traditional web automation difficult.
Execution of external scripts using cy.exec is not working as expected.
Questions:

Has anyone successfully integrated tools like AutoIT, SikuliX, or AutoHotkey with Cypress for similar scenarios?
Are there any best practices or considerations when using cy.exec for external script execution in this context?
How can I effectively handle synchronization and ensure the external script completes its actions within the Cypress test?
Additional Details:

Any insights, examples, or guidance you can provide would be greatly appreciated. Thank you in advance for your help!

**Comment by adnanerlansyah403** [comment]
It will be the case that I will getting in the future because my project also has the feature for uploading the image from my local machine. I hope this thread can keep it up
